src/LinearOperators.jl

```diff
@@ -823,4 +822,27 @@ function getindex(op :: AbstractLinearOperator,
   return R * op * E
 end
 
+"""`opBypass()`
+
+Bypass operator. Not to be mistaken with the opEye operator.
+```
+opBP = opBypass()
+v = rand(5)
+@assert opBP * v === v
+```
+"""
+struct opBypass <: AbstractLinearOperator{Any,Nothing,Nothing,Nothing} end
```

Comment:
This is really an identity operator, isn't it? Should we call the type itself `opEye` then?!

Reply:
It is an identity operator, but I made it different from an `opEye` because it doesn't work well with other operators: it doesn't have a size, can't be concatenated, can't be converted into a matrix, etc.

Reply:
Right, but `Bypass` is a bit of an unusual name. Could `opEye` be an instance of `Bypass` or something like that?

Reply:
Now that I think about, we can have it as `opEye`, because it'll be the only operator that doesn't accept arguments. So `opEye()` provides this functionality, but `opEye(n)` will do something else.

Reply:
Maybe that `opEye(n)` could be a preallocated `opEye`?

Reply:
I'm starting to think that we should have a separate type for operators with preallocation, and that is what Krylov.jl methods should take as input (including the preconditioner). What do you think?

Reply:
That's better. It defines clearly what to expect.